flink-clients/src/main/java/org/apache/flink/client/deployment/application/cli/ApplicationClusterDeployer.java

```diff
@@ -57,6 +60,16 @@ public <ClusterID> void run(
 
         LOG.info("Submitting application in 'Application Mode'.");
 
+        // In HA mode, we only support single-execute jobs at the moment. Here, we manually
+        // generate the job id, if not configured, to keep it consistent across failover.
+        if (HighAvailabilityMode.isHighAvailabilityModeActivated(configuration)
```

Comment:
Given that we just want a stable and unique job id for every Flink application, maybe we could also generate the job id based on `high-availability.cluster-id`.

And this could happen in the `ApplicationDispatcherBootstrap` of JobManager, not the client side.